File: evidence/generated-systems/angular/hr-on-angular/src/app/models/Timesheet.js

// Define collection and schema for Timesheet
export interface Timesheet {
    periodStart:
	type : Date
    periodEnd:
	type : Date
    submissionDate:
	type : Date
    Employee:
	type : Schema.Types.ObjectId
    TimeEntries:
 	type : [{ type: Schema.Types.ObjectId, ref: 'TimeEntry' }]
    Approvals:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Approval' }]
    Status:
 	type : String
#
    collection: 'timesheets'
}

File: evidence/generated-systems/rubyonrails/hr-on-rails/hronrails/app/models/timesheet.rb
class Timesheet < ApplicationRecord
  enum Status: [:Draft, :Submitted, :Approved, :Rejected, :Processed]


  has_many :Employee, class_name: 'Employee'
  has_many :TimeEntries, class_name: 'TimeEntry'
  has_many :Approvals, class_name: 'Approval'

end
